client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -90,6 +93,15 @@ class CalypsoifyIframe extends Component {
 
 			this.setState( { isMediaModalVisible: true, allowedTypes, gallery, multiple } );
 		}
+
+		if ( 'draftIdSet' === action && ! this.state.isDraftIdUpdated && ! this.props.postId ) {
+			const { postId } = payload;
+			const { currentRoute } = this.props;
+
+			this.props.replaceHistory( `${ currentRoute }/${ postId }`, true );
+			this.props.setRoute( `${ currentRoute }/${ postId }` );
+			this.setState( { isDraftIdUpdated: true } );
```

Comment:
is `replaceState` called multiple times on Gutenberg? Could this block other interactions, like trash?

Reply:
This is the only usage I've seen so far (setting the draft id):
https://github.com/WordPress/gutenberg/search?q=replaceState&unscoped_q=replaceState

Reply:
> Could this block other interactions, like trash?

If I'm not missing something, this shouldn't block other interactions since those will have different `action` set. 🤔 In addition to that, `replaceState` will still execute its default behavior prior to sending a message.

Reply:
I was mostly trying to answer two questions:

- do we need to keep state to prevent repetition?
- can we get away with a more generic action like `replaceHistory` instead of `draftIdSet` in case we have other interaction that needs to change the URL (not navigate away like trash, but trash was the first thing that I thought of).

Reply:
> do we need to keep state to prevent repetition?

I left it as a safety measure - in theory some unrelated functionality could modify the URL parameters, leaving `postId` unchanged, but we'd still transmit the action and append a valid id. This would lead to undesired state like `/post/siteSlug/1234/1234`. I have observed this while testing, but was not able to reproduce consistently (adding the state resolved it).

> * can we get away with a more generic action like `replaceHistory` instead of `draftIdSet` in case we have other interaction that needs to change the URL (not navigate away like trash, but trash was the first thing that I thought of).

No strong opinion here, but maybe we could generalize it if/when such needed arrises? The way I see it, we'll almost always need a separate action for that because I expect URL transformation to be different from the existing one.

Reply:
To avoid the never-ending slugs, you can also check the current url and see if it ends with the same postId you're trying to add.

https://github.com/Automattic/wp-calypso/blob/ea7a0ca72fc372d0752ce2e45cbb75dc6b0ed80c/client/gutenberg/editor/layout/browser-url/index.jsx#L32

Reply:
Updated in e7bd6345d7ed9f6120ca32b1eb855cb5d7a819f9.